### ISSUE TITLE
[Bug] VAOS Custom Dimensions Inherited in Previous Event

### ISSUE THREAD

**Issue by nedierecel** [department-of-veterans-affairs/va.gov-team#7093]
## What happened?
After a VAOS error, cancellation, or submission occurs, Google Analytics is inheriting the previous custom dimensions for an unrelated event. 

Example: I submitted a cancellation for VAOS, which sent the expected cancellation dataLayer values to Google Analytics. These custom dimensions were then inherited with a submission event.

![image.png](https://images.zenhubusercontent.com/5b295b9ff378ea1890b3d77e/8cabf2b1-5610-4914-bd10-bbbf9b1612e7)

## Steps to Reproduce
- URL: https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/
- Test User _(if applicable)_: Will provide over phone

## Desired behavior
_Describe in detail what was supposed to happen; reference existing Github issue that contains a solution if possible_

## Acceptance Criteria
- [ ] Identify cause of custom dimensions sent with unrelated events; Fix within Tag Manager or send back to FE VAOS team for resolution

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with `Bug`**

**Comment by nedierecel** [comment]
@narin I'm looping in my teammate @bsmartin-ep to review, as there still appears to be an issue where dataLayer values sent on previous events are inherited on the next event. I attempted a workaround in Google Tag Manager.

@bsmartin-ep To recreate, you can use the GTM preview link and user name I provided. This preview link has the most updated GTM settings.

**Comment by bsmartin-ep** [comment]
@narin - Is there an ETA for the staging system to be back online?

"Sorry, our system is temporarily down while we fix a few things. Please try again later."

Thanks.

**Comment by narin** [comment]
Hey @bsmartin-ep.  MVI has been down since yesterday and the VSP team has been working on getting that fixed.  I don't have an ETA unfortunately.  I can comment on this ticket as soon I find out it's back up.

**Comment by narin** [comment]
@bsmartin-ep Looks like staging is working now if you haven't tried recently

**Comment by bsmartin-ep** [comment]
@nedierecel -

I believe the issue is with this code in the VAOS app:

https://github.com/department-of-veterans-affairs/vets-website/blob/b38a5be6ec384428ca9c09607205bedbf49d47c6/src/applications/vaos/actions/appointments.js#L303-L306

The _appointmentType_ and _facilityType_ fields are updated in the datalayer with the "${eventPrefix}-successful" event, but I don't see anywhere where they are being reset.  When I go to schedule another appointment, they are still in the datalayer:

```
appointmentType: 'confirmed',
facilityType: 'va'
```

If you think these fields should still be attached to this _-successful_ event, then I would need to defer to the FE team for a fix.  Is there anywhere else they are resetting datalayer values after a transaction has  completed?

**Comment by nedierecel** [comment]
@narin - After further review, @bsmartin-ep uncovered why values are persisting. After a `-successful` event (whether a scheduled appointment or cancellation), can the dataLayer values be reset before the next `-successful` event?

**Comment by narin** [comment]
Thanks @bsmartin-ep @nedierecel. I'll have some time to look into this today.  May be reaching out with some questions.

**Comment by joanneesteban** [comment]
@narin - Thanks for looking into this! Do you have any updates for us?

**Comment by narin** [comment]
@joanneesteban @nedierecel @bsmartin-ep From what I understand, we need to reset the dimensions immediately after successful or failed submissions.  Is that correct?  I've created [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/12093) which resets the following dimensions.  Please let me know if this looks okay:

```
  recordEvent({
    flow: undefined,
    'health-TypeOfCare': undefined,
    'health-ReasonForAppointment': undefined,
    'error-key': undefined,
    appointmentType: undefined,
    facilityType: undefined,
  });
```

**Comment by bsmartin-ep** [comment]
Testing this in the browser console right now @narin.  I want to double check this will work and you don't need to do some weird array deletion instead.

**Comment by narin** [comment]
FYI, I followed the advice of this blog post: https://www.bounteous.com/insights/2016/03/21/removing-values-datalayer-data-layer-best-practices-pt-4/

**Comment by bsmartin-ep** [comment]
@narin - Ah - I think I was confusing it for when the _undefined_ is the name in the name/value pair.

Is this staged for testing?  (Sorry I'm new and am unfamiliar with your deployment process.)

**Comment by narin** [comment]
@bsmartin-ep The PR was approved already, and as soon as I merge it'll be deployed to staging.  We also have an automated deploy to production that starts at 2pm EST.  I'll wait until right after to merge so that we can test on staging first and the changes won't go live until tomorrow 2pm.

**Comment by narin** [comment]
@bsmartin-ep Changes have been deployed to staging. Let me know how it goes!

**Comment by bsmartin-ep** [comment]
LGTM @narin , but @nedierecel can you triple check?

Data layer looks clean in the events after `vaos-cancel-appointment-submission-successful`



...and I don't see any inheritance in `vaos-request-submission-successful`

**Comment by nedierecel** [comment]
Thanks @bsmartin-ep and @narin - things look good in my QA. Here are my steps:

1. I began with a submission-successful event:

![image.png](https://images.zenhubusercontent.com/5b295b9ff378ea1890b3d77e/bfa9df82-140a-4637-9e11-45e6329819d9)

2. Then, I cancelled an appointment and see the right values.
![image.png](https://images.zenhubusercontent.com/5b295b9ff378ea1890b3d77e/653e2512-422b-4da0-84f9-f773ce287873)
